processor/src/decoder/mod.rs

```diff
@@ -175,11 +186,23 @@ impl Process {
         self.execute_op(Operation::Noop)
     }
 
+    /// Continues decoding a SPAN block by absorbing the next batch of operations.
+    pub(super) fn respan(&mut self, op_batch: &OpBatch) {
+        self.decoder.respan(op_batch);
+
+        // send a request to the chiplets to continue the hash and absorb new elements.
+        self.chiplets.request_control_block_hash();
+    }
+
     /// Ends decoding a SPAN block.
     pub(super) fn end_span_block(&mut self, block: &Span) -> Result<(), ExecutionError> {
         // this appends a row with END operation to the decoder trace. when END operation is
         // executed the rest of the VM state does not change
         self.decoder.end_span(block.hash().into());
+
+        // send the end of control block to the chiplets bus to handle the final hash request.
+        self.chiplets.request_control_block_hash();
```

Comment:
I'm happy to change this, although that name doesn't match the current functionality (since currently it handles all lookups that don't initialize a hash, which includes `absorb` as well as `return`). I'll leave further comments in the more detailed discussion below.